lib/web_ui/dev/test_platform.dart

```diff
@@ -164,33 +182,61 @@ class BrowserPlatform extends PlatformPlugin {
   ''';
     }
     final List<int> goldenBytes = file.readAsBytesSync();
-    final int lengths = bytes.length;
-    for (int i = 0; i < lengths; i++) {
-      if (goldenBytes[i] != bytes[i]) {
-        if (write) {
-          file.writeAsBytesSync(bytes, flush: true);
-          return 'Golden file $filename was updated. You can remove "write: true" in the call to matchGoldenFile.';
-        } else {
-          final File failedFile = File(p.join(file.parent.path, '.${p.basename(file.path)}'));
-          failedFile.writeAsBytesSync(bytes, flush: true);
-
-          // TODO(yjbanov): do not fail Cirrus builds. They currently fail because Chrome produces
-          //                different pictures. We need to pin Chrome versions and use a fuzzy image
-          //                comparator.
-          if (Platform.environment['CIRRUS_CI'] == 'true') {
-            return 'OK';
+
+    // Migrate to flutter_test/lib/src/_goldens_io.dart ?
```

Comment:
I understand there's some duplication here with the work done in flutter_tools, but I really needed this to understand wtf was going on in some of the failures I was seeing, and I thought it'd be interesting to have (until we use the appropriate golden comparison code)

Reply:
I think this duplication is somewhat inevitable. flutter/engine can't depend on anything in flutter/flutter, so the only option would be to move it into a 3rd library. That would be a lot more work, especially when this is still quite new.

Reply:
Aggg, you're right. I guess I'd like to have the diffing shared so golden comparisons are consistent across all implementations, but I agree with you that moving only that to a 3rd party pkg diminishes returns big time!